src/lib.rs

```diff
@@ -96,6 +96,29 @@ where
     RetryPolicy::default().retry_if(task, condition).await
 }
 
+/// Retries a successful `Future` under a certain provided condition with a default `RetryPolicy`
+///
+/// ```
+/// again::retry_with(
```

Comment:
Let me stew over this for a bit. My initial reaction on naming is that this feels like a type of `fold` an initial value and a func to drive it to a single value. There might be an opportunity to drive this name closer to an analog from std lib

Reply:
Or some kind of `collect` I see it's collecting results from each invocation in its return type

Reply:
I think `collect` sounds really good and the initial idea at least is more to collect than to fold. And then the other function I was talking about, that handles both success conditions and error conditions could be `collect_and_retry`.